src/builder/code.rs

```diff
@@ -77,23 +76,31 @@ impl<F> SignatureBuilder<F> where F: Invoke<elements::FunctionType> {
 	}
 }
 
-impl<F> Invoke<Vec<elements::ValueType>> for SignatureBuilder<F>
```

Comment:
Any advice on how these should be updated @NikVolf? With the way things are, any `ValueType` would be added to params. Before, the `invoke` implementation was differentiated for `params` and `return_type` by the `Option` that `return_type` was contained in. Since `params` and `results` are the same type now it can't be done that way. I'm struggling to see how this should work...